src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -35,147 +30,133 @@ public partial class Regex : ISerializable
         protected internal string[]? capslist;                // if captures are sparse or named captures are used, this is the sorted list of names
         protected internal int capsize;                       // the size of the capture array
 
-        internal ExclusiveReference? _runnerref;              // cached runner
         internal WeakReference<RegexReplacement?>? _replref;  // cached parsed replacement pattern
-        internal RegexCode? _code;                            // if interpreted, this is the code for RegexInterpreter
-        internal bool _refsInitialized = false;
+        private ExclusiveReference? _runnerref;               // cached runner
+        private RegexCode? _code;                             // if interpreted, this is the code for RegexInterpreter
+        private bool _refsInitialized = false;
 
         protected Regex()
         {
             internalMatchTimeout = s_defaultMatchTimeout;
         }
 
         /// <summary>
-        /// Creates and compiles a regular expression object for the specified regular
-        /// expression.
+        /// Creates a regular expression object for the specified regular expression.
         /// </summary>
-        public Regex(string pattern) =>
-            Init(pattern, RegexOptions.None, s_defaultMatchTimeout, addToCache: false);
+        public Regex(string pattern) :
+            this(pattern, culture: null)
+        {
+        }
 
         /// <summary>
-        /// Creates and compiles a regular expression object for the specified regular
-        /// expression, and adds it to the cache.
+        /// Creates a regular expression object for the specified regular expression, with options that modify the pattern.
         /// </summary>
-        private Regex(string pattern, bool addToCache) =>
-            Init(pattern, RegexOptions.None, s_defaultMatchTimeout, addToCache);
+        public Regex(string pattern, RegexOptions options) :
+            this(pattern, options, s_defaultMatchTimeout, culture: null)
+        {
+        }
 
-        /// <summary>
-        /// Creates and compiles a regular expression object for the
-        /// specified regular expression with options that modify the pattern.
-        /// </summary>
-        public Regex(string pattern, RegexOptions options) =>
-            InitWithPossibleCompilation(pattern, options, s_defaultMatchTimeout, addToCache: false);
+        public Regex(string pattern, RegexOptions options, TimeSpan matchTimeout) :
+            this(pattern, options, matchTimeout, culture: null)
+        {
+        }
+
+        internal Regex(string pattern, CultureInfo? culture)
+        {
+            // Call Init directly rather than delegating to a Regex ctor that takes
+            // options to avoid rooting the Regex compiler unless necessary.
```

Comment:
I know this comment already existed, but I'm not sure I understand what it is saying. What does `avoid rooting the Regex compiler` mean?

Reply:
It's referring to the linker; I can make it more explicit.

If you only use `new Regex(string)`, then in theory the linker can see that there's no way you'd end up using the RegexCompiler and can trim it all away from your app.  But if `new Regex(string)` has a static code path that touches RegexCompiler, then it can't.

Reply:
Oh, I was thinking "rooting" like a GC/memory "root" which is why I was confused.

Reply:
I'll tweak the comment in my next PR.  It builds on this one so I'd like to get this in.  Thanks.